spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java

```diff
@@ -49,15 +51,20 @@ public UserService(UserRepository userRepository,RoleService roleService, Conver
         return result.map(converters::userToDTO);
     }
 
-    public Optional<UserDTO> saveUser(UserDTO userDTO) {
+    public Optional<UserDTO> saveUser(UserInsertionDTO userDTO) {
         User user=converters.DTOToUser(userDTO);
+        System.out.println(user.toString());
         val userAuth=Auth.getCurrentUser();
 
         if(!roleService.isAdmin(userAuth))
             if(!Auth.getAdd(userAuth))
                 return Optional.empty();
 
         user.setId(0L);
+        user.setPasswordHash(passwordEncoder.encode(userDTO.getPassword()));
+        user.setEnabled(true);
+        user.setAuthorities(Collections.singleton(roleService.employeeRole()));
+        user.setPermissions(Permission.builder().add(false).read(null).write(null).build());
```

Comment:
Extract this to corresponding converter.

Reply:
done